test/integration/kongaddon_test.go

```diff
@@ -0,0 +1,82 @@
+//go:build integration_tests
+// +build integration_tests
+
+package integration
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kongaddon "github.com/kong/kubernetes-testing-framework/pkg/clusters/addons/kong"
+	environment "github.com/kong/kubernetes-testing-framework/pkg/environments"
+)
+
+type customImageTest struct {
+	controllerImageRepo string
+	controllerImageTag  string
+	proxyImageRepo      string
+	proxyImageTag       string
+}
+
+func (tc customImageTest) controllerImage() string {
+	return fmt.Sprintf("%s:%s", tc.controllerImageRepo, tc.controllerImageTag)
+}
+
+func (tc customImageTest) proxyImage() string {
+	return fmt.Sprintf("%s:%s", tc.proxyImageRepo, tc.proxyImageTag)
+}
+
+func (tc customImageTest) name() string {
+	return fmt.Sprintf("KongAddonImages:[%s,%s]", tc.controllerImage(), tc.proxyImage())
+}
+
+func TestKongAddontWithCustomImage(t *testing.T) {
+	tests := []customImageTest{
+		{"kong/kubernetes-ingress-controller", "2.3.0", "kong", "2.7"},
+		{"kong/kubernetes-ingress-controller", "2.3.1", "kong", "2.8"},
```

Comment:
Would it be controversial from my side to suggest adding field names in place where struct initialization happens? This way it'd be obvious which fields are initialized with which values (without looking at struct definition)